routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
yeah... i'm not sure of a 'right' way to do this. but its really nice to be able to use the context for that purpose, allows you to easily register for just the notifications that you are interested in